AzureMonitor: Add resource-centric queries to datasource.ts (#32137)
grafana/grafana: docs/sources/datasources/azuremonitor.md

| -------------------------------------------------- | ------------------------------------------------------------------------------------------------------ |
| `workspaces()`                                     | Returns a list of workspaces for the default subscription.                                             |
| `workspaces(12345678-aaaa-bbbb-cccc-123456789aaa)` | Returns a list of workspaces for the specified subscription (the parameter can be quoted or unquoted). |
| `resources()`                                      | Returns a list of resources for the default subscription.                                              |

> | `resources(12345678-aaaa-bbbb-cccc-123456789aaa)`  | Returns a list of resources for the specified subscription (the parameter can be quoted or unquoted).  |

Reviewer: ```suggestion
| `resources(12345678-aaaa-bbbb-cccc-123456789aaa)`  | Returns a list of resources for the specified subscription (the parameter can be within quotes or it can be without quotes).  |
```

| `resources()`                                                                           | Returns a list of resources for default subscription      |
| `resources("12345678-aaaa-bbbb-cccc-123456789aaa")`                                     | Returns a list of resources for a specified subscription  |
| `resources("$subscription")`                                                            | With template variable for the subscription parameter     |
| `workspace("myWorkspace").Heartbeat \| distinct Computer`                               | Returns a list of Virtual Machines                        |
| `workspace("$workspace").Heartbeat \| distinct Computer`                                | Returns a list of Virtual Machines with template variable |
| `workspace("$workspace").Perf \| distinct ObjectName`                                   | Returns a list of objects from the Perf table             |
| `workspace("$workspace").Perf \| where ObjectName == "$object" \| distinct CounterName` | Returns a list of metric names from the Perf table        |

> | `resource("myResource").Heartbeat \| distinct Computer`                                 | Returns a list of Virtual Machines                        |

Reviewer: ```suggestion
| `resource("myResource").Heartbeat \| distinct Computer`                                 | Returns a list of virtual machines                        |
```

Author: Resolving these suggestions to be consistent with the description above.

> | `resource("$resource").Heartbeat \| distinct Computer`                                  | Returns a list of Virtual Machines with template variable |

Reviewer: ```suggestion
| `resource("$resource").Heartbeat \| distinct Computer`                                  | Returns a list of virtual machines with template variable |
```
